### DISCUSSION TITLE
Errors in terminal and can't install npm modules. Maybe `asdf`?

### DISCUSSION THREAD

**Discussion by ramiroaraujo** [johnlindquist/kit#1153] [Q&A]
Probably noob question, but I can't seem to have a proper terminal running in Kit. I immediately get an error when opening: `/opt/homebrew/opt/asdf/libexec/asdf.sh:.:35: no such file or directory: /Users/user/.asdf/lib/asdf.sh`

and when trying to install for example a dependency module from any script I get:
`npm i -D openai
Unknown command: asdf exec npm i -D openai`
`/opt/homebrew/opt/asdf/libexec/bin/asdf: line 82: /Users/user/.asdf/lib/commands/command-help.bash: No such file or directory`

I've tried the option to Sync path from terminal to Kit, but doesn't make a diff. I'll dig deeper later, but maybe someone's seen something similar or related?

Thanks!

**Comment by johnlindquist** [comment]
@ramiroaraujo That looks like something on your PATH is broken.

Can you "echo $PATH" in your regular terminal and paste it here?

You can adding/tweaking the PATH env var in ~/.kenv/.env to set the PATH for the terminal.

Does that make sense?

**Reply by ramiroaraujo** [reply]
Hmm, looks ok to me:
`/Users/ramiroaraujo/.asdf/shims:/opt/homebrew/opt/asdf/libexec/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/opt/homebrew/opt/fzf/bin`

**Reply by johnlindquist** [reply]
@ramiroaraujo If you copy/paste that to "PATH= paste here" in your ~/.kenv/.env, does the Kit terminal start working?

(To be able to install packages, you'll need ~/.knode/bin on that PATH too)


Also, sorry, I'm  on my phone. This stuff is hard to type out 😅

**Comment by ramiroaraujo** [comment]
@johnlindquist hmm not working. Just for fun I uninstalled and reinstalled, because I had played around with a few settings in the past. Same output:
![image](https://user-images.githubusercontent.com/46276/223127549-9b3e2712-ec44-4c75-ab46-344f6919e4b0.png)
And this is my .env:
![image](https://user-images.githubusercontent.com/46276/223127662-877ba76c-9d90-4bda-9cd8-84e7b619a602.png)
(I placed ~/.knode/bin first in the path)
And for reference my normal PATH:
![image](https://user-images.githubusercontent.com/46276/223127794-380a04e1-02ba-493a-9815-49efc5408328.png)
 
:|

**Reply by johnlindquist** [reply]
@ramiroaraujo 

1. Do you use nvm or any other scripts that get sourced when zsh is starting up?

The error reads like it's trying to find "node" somewhere on your PATH, but finding "asdf" instead. Your PATH looks good though, so my next thought is that zsh is sourcing something (in your .zshrc maybe?) that is interfering with the lookup.

2. Does "~/.knode/bin/node -v" work?

**Comment by ramiroaraujo** [comment]
`~/.knode/bin/node -v` works fine
Yes, I'm sourcing `asdf` in my `.zshrc`. TBH I don't _need_ `asdf`, though it's pretty useful for versioning different common runtimes. I'll try disabling it later and let you know

**Reply by ramiroaraujo** [reply]
by "sourcing" `asdf` I mean bootstrapping it

**Reply by johnlindquist** [reply]
K, I'll try installing asdf later today too. Thanks for all the info

**Reply by ramiroaraujo** [reply]
@johnlindquist actually `asdf` wasn't the issue. It was the `asdf` plugin for `oh-my-zsh`, or maybe a double bootstrap that might be happening 🤦‍♂️
That being said, I notice the homebrew bin dir is being removed from the path inside Kit. Is that expected behavior?

**Reply by johnlindquist** [reply]
@ramiroaraujo 

The term is launching zsh just like any other terminal we install does. I'll see if I can find why homebrew might be missing... Thx

**Reply by johnlindquist** [reply]
@ramiroaraujo 

#TIL ...
1. homebrew is added to the PATH in `~/.zprofile`
2. `~/.zprofile` is only loaded when we launch zsh in "login" mode

I've been able to launch with login mode, but it's "pre-pending" homebrew to the PATH.  I need to figure out how to make sure Kit node's is in front so that it uses the proper `npm` when installing packages.

> Note: This issue never surfaced during dev because I was launching the dev version in my terminal with `yarn dev` and it inherited the current shell's PATH 🤦‍♂️

Thanks a million for bringing this to my attention.
